File: file-service/src/main/java/com/architectcgz/file/common/exception/BusinessException.java
package com.architectcgz.file.common.exception;

import com.architectcgz.file.common.constant.FileServiceErrorCodes;

/**
 * 业务异常
 */
public class BusinessException extends RuntimeException {
    
    private final String code;
    
    public BusinessException(String message) {
        super(message);
        this.code = FileServiceErrorCodes.BUSINESS_ERROR;
    }
    
    public BusinessException(String code, String message) {
        super(message);
        this.code = code;
    }

    public BusinessException(String message, Throwable cause) {
        super(message, cause);
        this.code = FileServiceErrorCodes.BUSINESS_ERROR;
    }

    public BusinessException(String code, String message, Throwable cause) {
        super(message, cause);
        this.code = code;
    }

    public String getCode() {
        return code;
    }

    public String getErrorCode() {
        return code;
    }
}

File: backend/Exceptions/BusinessException.cs
using FileService.Constants;

namespace FileService.Exceptions;

public class BusinessException(int code, string errorMessage) : Exception(errorMessage)
{
    public int Code { get; set; } = code;
    public string ErrorMessage { get; set; } = errorMessage;

    public BusinessException(BusinessError businessError) 
        : this(businessError.Code, businessError.Message)
    {
    }
}

